content/en/platform/corda/5.0/developing-applications/packaging/cpb.md

```diff
@@ -0,0 +1,89 @@
+---
+date: '2023-02-23'
+title: "Build a CPB"
+project: corda
+version: 'Corda 5.0'
+menu:
+  corda5:
+    parent: corda5-develop-packaging
+    identifier: corda5-develop-packaging-cbp
+    weight: 2000
+section_menu: corda5
+---
+You can build a CPB from your CPK files using one of the following:
+* [Gradle]({{< relref "#build-a-cpb-using-gradle" >}})
+* [Corda CLI]({{< relref "#build-a-cpb-using-the-corda-cli" >}})
+
+### Build a CPB Using Gradle
+
+To build a CPB using Gradle:
+
+1. Add the CPB plugin to your project by adding the following to the start of the `build.gradle` file of your CorDapp Gradle project:
+   ```
+   plugins {
+       id 'net.corda.plugins.cordapp-cpb2'
+   }
+   ```
+
+2. To sign your CPB, add a `cordapp/signing` section to the project `settings.gradle` file.
```

Comment:
@josephzunigadaly Should we add the example config from here: https://github.com/corda/corda-gradle-plugins/tree/master/cordapp-cpk2#dsl Or is it better not to go into specifics for now?

Reply:
Here is an example:
```
    signing {
        options {
            alias = "my-signing-key"
            storePassword = "keystore password"
            keyStore = layout.projectDirectory.dir("../workspace").file("signingkeys.pfx").toString()
            storeType = "PKCS12"
        }
    }
```
This example was built in `workflows/build.gradle` of https://github.com/corda/CSDE-cordapp-template-kotlin

Adding this to the CSDE project causes the build CPI step to fail. This ticket will fix it: https://r3-cev.atlassian.net/browse/CORE-7275
Might be worth raising a docs ticket with this example and adding a blocked type link to CORE-7275 so we can come back and add this once it is fixed.
We should test the example in the CSDE project once it is fixed too.

Reply:
https://r3-cev.atlassian.net/browse/DOC-5314 linked and generated. This will be updated then.